### ISSUE TITLE
ListView: Talkback can't reach offscreen elements if a view is pushed and then popped

### ISSUE THREAD

**Issue by lorenzOliveto** [flutter/flutter#40419]
## Steps to Reproduce

With talkback on:
1. Create a `ListView` with some elements 
2. Push a new Widget using `MaterialPageRoute` and then pop it
3. Elements not visible on the screen cannot be reached by talkback.

Tried on a Nexus 5x.
On iOS/Voiceover it seems to work correctly.

Here's a basic example:

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Material App',
      theme: ThemeData.dark(),
      home: MyHomePage(title: 'how are you',),
    );
  }
}
class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: ListView.builder(
        itemCount: 100,
        itemBuilder: (context, index) {
          return ListTile(
            title: Text('Item $index'),
            onTap: () {
              Navigator.of(context).push(
                MaterialPageRoute(
                  builder: (context) => PushedPage(),
                ),
              );
            },
          );
        },
      ),
    );
  }
}

class PushedPage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Pushed page'),
      ),
    );
  }
}


```
## Flutter analyze output

```
Analyzing flutter_app1...                                               
No issues found! (ran in 14.4s)
```
## Flutter doctor output

```
[✓] Flutter (Channel stable, v1.9.1+hotfix.2, on Mac OS X 10.14.6 18G95, locale it-IT)
    • Flutter version 1.9.1+hotfix.2 at /Users/lorenzooliveto/Flutter/flutter
    • Framework revision 2d2a1ffec9 (6 days ago), 2019-09-06 18:39:49 -0700
    • Engine revision b863200c37
    • Dart version 2.5.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/lorenzooliveto/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 10.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.3, Build version 10G8
    • CocoaPods version 1.7.5

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 39.0.3
    • Dart plugin version 191.8423
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

 
[✓] Connected device (1 available)            
    • Nexus 5X • 00953f3bb077bb69 • android-arm64 • Android 8.1.0 (API 27)

• No issues found!
```

**Comment by TahaTesser** [comment]
Hi @lorenzOliveto 
Are you still experiencing this issue?
Thank you

**Comment by lorenzOliveto** [comment]
Hi @TahaTesser,
Yes I'm still experiencing this issue, basically once the pushed view is popped talkback can't focus on elements not visible in the ListView and jumps to the top instead of scrolling the list.

Here's my Flutter doctor output:

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, v1.12.13+hotfix.8, on Mac OS X 10.15.3 19D76, locale it-IT)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
[✓] Android Studio (version 3.6)
[✓] Connected device (1 available)

• No issues found!
```

**Comment by markusaksli-nc** [comment]
Reproducible on the latest master `1.22.0-2.0.pre.59`.
<details>
 <summary>Minimal reproducible code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MaterialApp(
    home: MyHomePage(),
    title: "Sample",
  ));
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text("Sample"),
      ),
      body: ListView.builder(
        itemCount: 100,
        itemBuilder: (context, index) {
          return ListTile(
            title: Text('Item $index'),
            onTap: () {
              Navigator.of(context).push(
                MaterialPageRoute(
                  builder: (context) => PushedPage(),
                ),
              );
            },
          );
        },
      ),
    );
  }
}

class PushedPage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Pushed page'),
      ),
    );
  }
}
```
</details>
<details>
 <summary>flutter doctor -v</summary>

```
[√] Flutter (Channel master, 1.22.0-2.0.pre.59, on Microsoft Windows [Version 10.0.19041.450], locale en-US)
    • Flutter version 1.22.0-2.0.pre.59 at C:\Development\flutter_master
    • Framework revision be7c7faee6 (7 hours ago), 2020-08-24 22:31:02 -0400
    • Engine revision 8107b36c7b
    • Dart version 2.10.0 (build 2.10.0-56.0.dev)

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.1)
    • Android SDK at C:\Users\marku\AppData\Local\Android\sdk
    • Platform android-30, build-tools 30.0.1
    • Java binary at: C:\Users\marku\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\193.6626763\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.6.5)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.6.30320.27
    • Windows 10 SDK version 10.0.18362.0

[√] Android Studio (version 4.0)
    • Android Studio at C:\Users\marku\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\193.6626763
    • Flutter plugin version 47.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] Connected device (4 available)
    • sdk gphone x86 arm (mobile) • emulator-5554 • android-x86    • Android 11 (API 30) (emulator)
    • Windows (desktop)           • windows       • windows-x64    • Microsoft Windows [Version 10.0.19041.450]
    • Web Server (web)            • web-server    • web-javascript • Flutter Tools
    • Chrome (web)                • chrome        • web-javascript • Google Chrome 84.0.4147.135

• No issues found!
```
</details>

**Comment by xster** [comment]
Internal issue b/193871309. @chunhtai could you help take a look at this GAR4 issue?

**Comment by chunhtai** [comment]
The implicit scroll flag was gone after the page transition, not sure what happened, investigating

**Comment by goderbauer** [comment]
This fix is currently blocked on the scroll metrics PR to roll into google3.
